src/mdrv2.cpp

```diff
@@ -117,12 +122,104 @@ void updatePcieInfo(UpdateDBusData infoPcieDevs) {
 
     for (auto& pciDev : pcieDevices)
     {
-        pciDev.second->pcieInfoUpdate();
+        pciDev.second->pcieInfoUpdate(motherboardPath);
+    }
+}
+
+void getMotherboardPath(sdbusplus::bus_t& bus, const std::string& inventoryPath,
+                        std::string& motherboardPath)
+{
+    // By default, look for System interface on any system/board/* object
+    std::string mapperAncestorPath = inventoryPath;
+    std::string matchParentPath = inventoryPath + "/board/";
+    bool requireExactMatch = false;
+
+    if (inventoryPath != defaultInventoryPath)
+    {
+        std::filesystem::path path(inventoryPath);
+
+        // Search under parent to find exact match for self
+        mapperAncestorPath = path.parent_path().string();
+        matchParentPath = mapperAncestorPath;
+        requireExactMatch = true;
+    }
+
+    auto method = bus.new_method_call(mapperBusName, mapperPath,
+                                      mapperInterface, "GetSubTreePaths");
+    method.append(mapperAncestorPath);
+    method.append(0);
+    method.append(std::vector<std::string>({systemInterface}));
+
+    try
+    {
+        std::vector<std::string> paths;
+        sdbusplus::message_t reply = bus.call(method);
+        reply.read(paths);
+
+        size_t pathsCount = paths.size();
+        for (size_t i = 0; i < pathsCount; ++i)
+        {
+            if (requireExactMatch && (paths[i] != inventoryPath))
+            {
+                continue;
+            }
+
+            motherboardPath = std::move(paths[i]);
+            break;
+        }
+        if (motherboardPath.empty())
+        {
+            phosphor::logging::log<phosphor::logging::level::ERR>(
+                "Failed to get system motherboard dbus path. Setting up a "
+                "match rule");
+
+            if (!motherboardConfigMatch)
+            {
+                motherboardConfigMatch =
+                    std::make_unique<sdbusplus::bus::match_t>(
+                        bus,
+                        sdbusplus::bus::match::rules::interfacesAdded() +
+                            sdbusplus::bus::match::rules::argNpath(
+                                0, matchParentPath),
+                        [&](sdbusplus::message_t& msg) {
+                    sdbusplus::message::object_path objectName;
+                    boost::container::flat_map<
+                        std::string,
+                        boost::container::flat_map<
+                            std::string, std::variant<std::string, uint64_t>>>
+                        msgData;
+                    msg.read(objectName, msgData);
+                    if (msgData.contains(systemInterface))
+                    {
+                        getMotherboardPath(bus, inventoryPath, motherboardPath);
```

Comment:
I am not sure with this.

Reply:
Can you please investigate this? It is your PR, not mine. You need to be fully sure about your code, and not just copy-paste things from `smbios-mdr`

Reply:
The flow is revised as follows:
1. Set up match rule in main(). Which is legitimate practice (see D-bus sensors for example).
2. Once rule is met call updateMappingsFromFile() on the second iteration and NOT in the context of the match handler.

The revision was necessary because previous flow resulted in core dump.
The previous flow:
1. Call updateMappingsFromFile()
2. Set up match rule from within updateMappingsFromFile()
3. Once rule is met, call updateMappingsFromFile() FROM the context of the match handler. Core dump as a result.